### DISCUSSION TITLE
how to override config with define_asset_job in dbt?

### DISCUSSION THREAD

**Discussion by dduong1603** [dagster-io/dagster#18803] [Q&A] [answered]
Is it possible to only override some config in define_asset_job? Or do I have to provide the full config schema?
If I'm defining my job like this
```python
dbt_job = define_asset_job(
    name="dbt_job",
    selection=(some_asset_selection),
    config={
        "resources": {
            "dbt": {
                "config": {
                    "profile": "some_other_profile",
                }
            }
        }
    },
)
```
I got
```
dagster._core.errors.DagsterInvalidConfigError: Invalid default_value for Field.
    Error 1: Missing required config entry "project_dir" at path root:dbt:config. Sample config for missing entry: {'project_dir': '...'}
```

**Comment by rexledesma** [comment, marked as answer]
You'll have to provide the full, valid config schema that you want in the override. I also recommend that you use [`RunConfig`](https://docs.dagster.io/concepts/resources#providing-resource-launch-time-configuration-in-python-code) when supplying this information.

For example, if you're using `DbtCliResource`, `project_dir` must be specified alongside your overrides.

```python
dbt_job = define_asset_job(
    name="dbt_job",
    selection=AssetSelection.all(),
    config=RunConfig(
        resources={
            "dbt": DbtCliResource(
                project_dir=os.fspath(dbt_project_dir),
                profile="dev",  # My overrides,
            )
        }
    ),
)
```

**Reply by dduong1603** [reply]
ahh so it's what I suspected. Would have been nice to just override the specific config though haha. Thanks @rexledesma
